app/lib/system/deploy.rb

```diff
@@ -4,28 +4,48 @@ module System
   module Deploy
     mattr_accessor :info
 
+    DEFAULT_DEPLOY_INFO_PATH = '.deploy_info'
+    private_constant :DEFAULT_DEPLOY_INFO_PATH
+
     # Provides information about the version of the code:
     #   - revision: low-level version, that is used by BugSnag as app version
     #   - release: customer-facing version, that is visible in the admin portal footer
     #   - deployed_at: timestamp of the deployment
-    # The data is taken from the `.deploy_info` file in the root directory, but for SaaS the release
-    # is overridden with VERSION.
+    #   - docs_version: product name and the version that are used as prefix for product documentation path in the
+    #                   Red Hat Customer Portal
+    # The data is taken from the `.deploy_info2` file in the root directory, but for SaaS the release
+    # is overridden with DEFAULT_VERSION.
+    # `.deploy_info2` file is injected to the container during container build process in CPaaS (the content is set in Dockerfile)
     # The information is exposed via {MASTER_PORTAL}/deploy endpoint for logged-in users.
     class Info
-      VERSION = '2.x'
-      private_constant :VERSION
+      DEFAULT_VERSION = '2.x'
+
+      private_constant :DEFAULT_VERSION
 
       attr_reader :revision, :deployed_at, :release
 
       delegate :minor_version, :major_version, to: :version
 
       def initialize(info)
         @revision = info.fetch('revision') { `git rev-parse HEAD 2> /dev/null`.strip }
-        @release = ThreeScale.config.onpremises ? info.fetch('release', VERSION) : VERSION
+        @release = saas? ? DEFAULT_VERSION : info.fetch('release', DEFAULT_VERSION)
         @deployed_at = info.fetch('deployed_at') { Time.now }
         @error = info.fetch(:error) if info.key?(:error)
       end
 
+      def docs_version
+        @docs_version ||= saas? || rhoam? ? 'red_hat_3scale/2-saas' : "red_hat_3scale_api_management/#{major_version}.#{minor_version}"
+      end
+
+      def saas?
```

Comment:
CodeClimate complained, but I think it's still worth it:
```
System::Deploy::Info#saas? doesn't depend on instance state (maybe move it to another class?) 
```

Reply:
not saying where this should be but just saying that we shouldn't always listen to static analysis

Reply:
> we shouldn't always listen to static analysis

:100: